src/promise-pool-executor.ts

```diff
@@ -430,14 +431,24 @@ export class PromisePoolExecutor<T, R> implements UsesConcurrency, Stoppable, St
     return this
   }
 
+  private areItemsValid (): boolean {
+    const items = this.items() as any
```

Comment:
I checked my `Symbol.iterator in x` idea: it doesn’t work for strings. Let’s keep this `as any` cast for now so that we’re fine for iterators using the square bracket notation 👍